### PR TITLE
Update the eo3 type for DE Africa to something custom

### PR DESCRIPTION
**Pull request by alexgleith**
* Add a bunch of custom fields
* Remove a bunch of Australia-specific custom fields.

### CODE REVIEW

File: metadata/eo3_deafrica.odc-type.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    # Custom DE Africa Fields
+    # Generic
+    projection:
+      description: Platform code
+      offset: ['properties', 'proj:epsg']
+
+    # Sentinel-1
+    sat_orbit_state:
+      description: The satellite orbit state, either ascending or descending
+      offset: ['properties', 'sat:orbit_state:']
```
Comment (fangfy) [inline metadata/eo3_deafrica.odc-type.yaml:75]: extra ":" after orbit_state?

File: metadata/eo3_deafrica.odc-type.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    
+    sat_relative_orbit:
+      description: The satellite relative orbit
+      offset: ['properties', 'sat:relative_orbit:']
```
Comment (fangfy) [inline metadata/eo3_deafrica.odc-type.yaml:79]: extra ":"

File: metadata/eo3_deafrica.odc-type.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      type: double
```
Comment (fangfy) [inline metadata/eo3_deafrica.odc-type.yaml:80]: does this mean it's a number? I wander if this field has value starting with 0 (e.g. 086). If so, this is probably easier to handle as a string?

Comment (fangfy) [reply metadata/eo3_deafrica.odc-type.yaml:80]: I'm actually not sure without looking at the STAC files.

File: metadata/eo3_deafrica.odc-type.yaml (hunk 1 above)
Comment (alexgleith) [reply metadata/eo3_deafrica.odc-type.yaml:99]: OOps, yep.

File: metadata/eo3_deafrica.odc-type.yaml (hunk 3 above)
Comment (alexgleith) [reply metadata/eo3_deafrica.odc-type.yaml:104]: Numbers were like `123`. We can handle as a string though, yep.

**Pull request merged**